spec/mysql2/client_spec.rb

```diff
@@ -345,127 +345,121 @@ def connect *args
+      expect(@client.query("SELECT 1", :symbolize_keys => true).first.keys[0].class).to eql(Symbol)
     end
 
     it "should require an open connection" do
       @client.close
-      lambda {
+      expect {
         @client.query "SELECT 1"
-      }.should raise_error(Mysql2::Error)
+      }.to raise_error(Mysql2::Error)
     end
 
     if RUBY_PLATFORM !~ /mingw|mswin/
       it "should not allow another query to be sent without fetching a result first" do
         @client.query("SELECT 1", :async => true)
-        lambda {
+        expect {
           @client.query("SELECT 1")
-        }.should raise_error(Mysql2::Error)
+        }.to raise_error(Mysql2::Error)
       end
 
       it "should describe the thread holding the active query" do
         thr = Thread.new { @client.query("SELECT 1", :async => true) }
 
         thr.join
-        begin
-          @client.query("SELECT 1")
-        rescue Mysql2::Error => e
-          message = e.message
-        end
-        re = Regexp.escape(thr.inspect)
-        message.should match(Regexp.new(re))
+        expect { @client.query('SELECT 1') }.to raise_error(Mysql2::Error, Regexp.new(Regexp.escape(thr.inspect)))
       end
 
       it "should timeout if we wait longer than :read_timeout" do
-        client = Mysql2::Client.new(DatabaseCredentials['root'].merge(:read_timeout => 1))
-        lambda {
-          client.query("SELECT sleep(2)")
-        }.should raise_error(Mysql2::Error)
-      end
-
-      if !defined? Rubinius
-        # XXX this test is not deterministic (because Unix signal handling is not)
-        # and may fail on a loaded system
-        it "should run signal handlers while waiting for a response" do
-          mark = {}
-          trap(:USR1) { mark[:USR1] = Time.now }
-          begin
-            mark[:START] = Time.now
-            pid = fork do
-              sleep 1 # wait for client "SELECT sleep(2)" query to start
-              Process.kill(:USR1, Process.ppid)
-              sleep # wait for explicit kill to prevent GC disconnect
-            end
-            @client.query("SELECT sleep(2)")
-            mark[:END] = Time.now
-            mark.include?(:USR1).should be_true
-            (mark[:USR1] - mark[:START]).should >= 1
-            (mark[:USR1] - mark[:START]).should < 1.3
-            (mark[:END] - mark[:USR1]).should > 0.9
-            (mark[:END] - mark[:START]).should >= 2
-            (mark[:END] - mark[:START]).should < 2.3
-            Process.kill(:TERM, pid)
-            Process.waitpid2(pid)
-          ensure
-            trap(:USR1, 'DEFAULT')
+        client = Mysql2::Client.new(DatabaseCredentials['root'].merge(:read_timeout => 0))
+        expect {
+          client.query('SELECT SLEEP(0.1)')
+        }.to raise_error(Mysql2::Error)
+      end
+
+      # XXX this test is not deterministic (because Unix signal handling is not)
+      # and may fail on a loaded system
+      it "should run signal handlers while waiting for a response" do
```

Comment:
Right, so that test actually passed on a previous version of rubinius (2.2.2?), which is why I removed the guard. Once the segfault is fixed (it might already be, current rbx is 2.2.10 but travis runs 2.2.5), the test will pass, so I think we should leave the guard out.